File: c/3783.c
int reverse(int n) {
    int res = 0;
    while (n != 0) {
        res = (res * 10) + (n % 10);
        n /= 10;
    }
    return res;
}

int mirrorDistance(int n) {
    int res = reverse(n) - n;
    if (res < 0) {
        return -res;
    }
    return res;
}

File: py/3783.py
class Solution:
    def reverse(self, n:int):
        if n < 10:
            return n
        return (n % 10) * (10 ** (len(str(n)) - 1)) + self.reverse(n // 10)

    def mirrorDistance(self, n: int) -> int:
        return abs(self.reverse(n) - n)
